### PR TITLE
Test driven

### PR DESCRIPTION
**Pull request by Kaghenimbale**
- [ ] Create a class called Solver.
- [ ] Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- [ ] Create a method called reverse that takes one argument, a string word, and returns the word reversed (e.g. if word is "hello" it returns "olleh").
- [ ] Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. 
- [ ] Add tests for all of the methods described above.

### CODE REVIEW

File: spec/factorial_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      solver = Solver.new
+      expect(solver.factorial(3)).to eql(6)
+    end
+
+    it 'return factorial of n' do
+      solver = Solver.new
+      expect(solver.factorial(5)).to eql(120)
+    end
+  end
+end
```
Comment (GzimAsani) [inline spec/factorial_spec.rb]: - [ ] Please add a negative case where the test should raise an exception ```raise_exception```
image

Comment (GzimAsani) [reply spec/factorial_spec.rb]: - [ ] Please add a test case when the N is 0 which is a special case and it should return 1
image

Comment (GzimAsani) [reply spec/factorial_spec.rb]: - [ ] Please add a test case when the N is 0 which is a special case and it should return 1

File: spec/fizzbuzz_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  describe '#fizzbuzz' do
+    it 'it should return fizz' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(6)).to eql('fizz')
+    end
+
+    it 'it should return buzz' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(10)).to eql('buzz')
+    end
```
Comment (GzimAsani) [inline spec/fizzbuzz_spec.rb]: - [ ] The numbers here need to be 3 and 5, please update them according to the requirements 
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @Kaghenimbale  and @danutibe07                :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with the tests, they're all passing, you just have few small change to make :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @Kaghenimbale,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**